eva-lib/src/main/java/uk/ac/ebi/eva/lib/utils/RepositoryFilter.java

```diff
@@ -0,0 +1,139 @@
+package uk.ac.ebi.eva.lib.utils;
+
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+
+import uk.ac.ebi.eva.lib.repository.VariantEntityRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RepositoryFilter<T> {
```

Comment:
This would be better modeled as one class per filter:

* The field would be set in the constructor, making the filter itself responsible to know its name (instead of an external class).
* It would be trivial to create new filters, being a new class with a very specific interface instead of modifying several methods in a single class.
* There could be multiple sub-hierarchies, to ensure that only `VariantRepositoryFilter` objects are passed to the `VariantEntityRepository`, another for studies and so on.